### PR TITLE
prow: tide: improve logging to allow for debugging

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/area  prow
/kind bug
/assign @cjwagner 
/cc @fejta

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@cjwagner added more logging to surround your new filtering code. We need to ensure that with debug level logging, it is possible to determine why each PR is in the pool that it is in (or is not). Otherwise, debugging Tide deployments is impossible.

File: prow/cmd/tide/main.go [hunk 1]
```diff
@@ -107,6 +107,9 @@ func main() {
 	if o.dryRun {
 		ghcSync = github.NewDryRunClient(oauthSecret, o.githubEndpoint.Strings()...)
 		ghcStatus = github.NewDryRunClient(oauthSecret, o.githubEndpoint.Strings()...)
+		if o.deckURL == "" {
```
Comment (fejta) [inline prow/cmd/tide/main.go]: Ideally we refactor this so that it is unit-testable, like we've done in peribolos, etc

**Comment by stevekuznetsov** [comment]
Upside is since it is not required anymore, `tide` should not care? But branch protection may not allow the merge

**Pull request merged**